### PR TITLE
Add user contribution icon and tooltip

### LINKED ISSUE #3439: Indicate next to a user's name on the site if they've classified requests
With a badge or some other sign of recognition

### PR DESCRIPTION
**Pull request by wrightmartin**
Here's the code to add these to a user's name. Needs dev now

```
<a href="#" class="has-tooltip">
      <%= image_tag "contribution-medal.png", :alt => "This user is contributor to the site" %>
      <span class="tooltip">
          <%= _('Valued contributor') %>
      </span>
    </a>
```

The `<a>` tag should link to the request game

<s>

```erb
<span data-tooltip aria-haspopup="true" class="js-tooltip has-tooltip" data-tooltip-content-id="tooltip_id_2nd_example"
data-tooltip-text="Valued contributor"
data-tooltip-prefix-class="alaveteli"
data-tooltip-link="http://www.google.com">
<%= image_tag "contribution-medal.png", :alt => "This user is contributor to the site" %>
</span>
```
</s>
Fixes #3439

### CODE REVIEW

**Comment by wrightmartin** [comment]
#3439

File: app/views/request/show.html.erb [hunk 1]
```diff
@@ -86,6 +86,15 @@
   <% end %>
 
   <p class="subtitle">
+    Robin Houston
+    <span data-tooltip aria-haspopup="true" class="js-tooltip has-tooltip" data-tooltip-content-id="tooltip_id_2nd_example"
+ data-tooltip-text="Valued contributor"
+ data-tooltip-prefix-class="alaveteli"
+ data-tooltip-link="http://www.google.com"
+ >
+ <%= image_tag "contribution-medal.png", :alt => "This user is contributor to the site" %>
```
Comment (garethrees) [inline app/views/request/show.html.erb]: Is the wording correct here? I thought we were showing the medal if they'd played the classification game? This matters in terms of the logic we need for showing the medal.

File: app/views/request/show.html.erb [hunk 2]
```diff
@@ -86,6 +86,15 @@
   <% end %>
 
   <p class="subtitle">
+    Robin Houston
+    <span data-tooltip aria-haspopup="true" class="js-tooltip has-tooltip" data-tooltip-content-id="tooltip_id_2nd_example"
+ data-tooltip-text="Valued contributor"
+ data-tooltip-prefix-class="alaveteli"
+ data-tooltip-link="http://www.google.com"
```
Comment (garethrees) [inline app/views/request/show.html.erb]: what should this link to?

File: app/views/request/show.html.erb [hunk 3]
```diff
@@ -86,6 +86,15 @@
   <% end %>
 
   <p class="subtitle">
+    Robin Houston
+    <span data-tooltip aria-haspopup="true" class="js-tooltip has-tooltip" data-tooltip-content-id="tooltip_id_2nd_example"
+ data-tooltip-text="Valued contributor"
+ data-tooltip-prefix-class="alaveteli"
```
Comment (garethrees) [inline app/views/request/show.html.erb]: should this be the sitename?

File: app/views/request/show.html.erb [hunk 4]
```diff
@@ -86,6 +86,15 @@
   <% end %>
 
   <p class="subtitle">
+    Robin Houston
+    <span data-tooltip aria-haspopup="true" class="js-tooltip has-tooltip" data-tooltip-content-id="tooltip_id_2nd_example"
```
Comment (garethrees) [inline app/views/request/show.html.erb]: "2nd_example"?

File: app/assets/stylesheets/responsive/_global_style.scss [hunk 5]
```diff
@@ -237,3 +237,53 @@ div.pagination {
...
+  }
+}
+
+.alaveteli-tooltip__close {
+  display: none; //temporary until we get around to styling it
+}
+
+.simple-tooltip-tooltip__close {
+  float: right;
+  /** fix typo inputs **/
```
Comment (garethrees) [inline app/assets/stylesheets/responsive/_global_style.scss]: I don't understand this comment – maybe some extra detail in the commit message would be good for explanation?

File: app/assets/javascripts/general.js [hunk 6]
```diff
@@ -42,6 +42,163 @@ $('#standard-popup .js-popup__close').click(function() {
...
+         if ( !event.shiftKey && (focusedItemIndex == numberOfFocusableItems - 1) ){
+             focusableItems.get(0).focus();
+             event.preventDefault();
+            }
+         if ( event.shiftKey && focusedItemIndex == 0 ){
+             focusableItems.get(numberOfFocusableItems - 1).focus();
+             event.preventDefault();
+            }
+
+
```
Comment (garethrees) [inline app/assets/javascripts/general.js]: Should be 2-space indent and lots of unnecessary blank lines.

File: app/assets/javascripts/general.js [hunk 7]
```diff
@@ -42,6 +42,163 @@ $('#standard-popup .js-popup__close').click(function() {
   return false;
 });
 
+// tooltips
+$(document).ready(function(){
+
+   /*
+    * jQuery simple and accessible dialog tooltip window, using ARIA
```
Comment (garethrees) [inline app/assets/javascripts/general.js]: This looks like it should live in its own file, with just the library configuration in `general.js`?

File: app/views/request/show.html.erb (hunk 4 above)
Comment (wrightmartin) [inline app/views/request/show.html.erb]: copy and paste from the plugin example code, will fix

File: app/assets/stylesheets/responsive/_global_style.scss (hunk 5 above)
Comment (wrightmartin) [inline app/assets/stylesheets/responsive/_global_style.scss]: This is code from the plugin itself - I don't know the detail

File: app/assets/javascripts/general.js (hunk 6 above)
Comment (wrightmartin) [inline app/assets/javascripts/general.js]: Yeah copy and paste error, will fix

**Comment by wrightmartin** [comment]
screen shot 2016-11-29 at 15 53 51

On click
screen shot 2016-11-29 at 15 53 56

**Comment by wrightmartin** [comment]
** UPDATE: I have rewritten this **

screen shot 2016-11-30 at 12 34 07
screen shot 2016-11-30 at 12 33 26

**Comment by garethrees** [comment]
Thanks @wrightmartin! This now needs development work to show the contribution medal around the site.

**Comment by garethrees** [comment]
This will need updating, so closing the PR for now. Will reopen when we revisit 
https://github.com/mysociety/alaveteli/issues/3439

**Pull request closed without merging**